references/validationFunctions/treatment/checkWhenNoTreatment.js

```diff
@@ -5,42 +5,82 @@
  */
 
-const validation = () => 
-  (function validate(inputs) {
-      const {$row, $name, $field} = inputs;
-      let result = {valid: true, message: "Ok"};
-      const coreFields = ['treatment_start_interval', 'treatment_duration', 'is_primary_treatment', 'treatment_intent', 'treatment_setting', 'response_to_treatment_criteria_method', 'response_to_treatment'];
- 
-      // checks for a string just consisting of whitespace
-      const checkforEmpty = (entry) => {return /^\s+$/g.test(decodeURI(entry).replace(/^"(.*)"$/, '$1'))};
-      const treatmentType = ($row.treatment_type).map(value => value.toLowerCase());
-       
-      if (!treatmentType.includes("no treatment") && coreFields.includes($name) && (!$field || $field === null || checkforEmpty($field))) {
-          result = { valid: false, message: `The '${$name}' field must be submitted when the 'treatment_type' field is '${treatmentType}'`};
-      }
-      else if (treatmentType.includes("no treatment") && ($field && $field != null && !(checkforEmpty($field)))) {
-        if (coreFields.includes($name) || (typeof($field) === 'string' && $field.trim().toLowerCase() != 'not applicable') || typeof($field) === 'number') {
-          result = { valid: false, message: `The '${$name}' field cannot be submitted if the 'treatment_type' field is '${treatmentType}'`};
+const validation = () =>
+    (function validate(inputs) {
+        const { $row, $name, $field } = inputs;
+        const result = { valid: true, message: 'Ok' };
+
+        const arrayItemsInSecondArray = (arr1, arr2) => {
+            return arr2.some(arr2Item => {
+                return arr1.includes(arr2Item);
+            });
+        };
+
+        const coreFields = [
+            'treatment_start_interval',
+            'treatment_duration',
+            'is_primary_treatment',
+            'treatment_intent',
+            'treatment_setting',
+            'response_to_treatment_criteria_method',
+            'response_to_treatment',
+        ];
+
+        const treatmentExceptionTypes = ['no treatment', 'unknown'];
+
+        // checks for a string just consisting of whitespace
+        const checkforEmpty = entry => {
+            return /^\s+$/g.test(decodeURI(entry).replace(/^"(.*)"$/, '$1'));
+        };
+        const treatmentTypes = $row.treatment_type.map(value => value.toLowerCase());
+
+        const treatmentTypesAreValid = !arrayItemsInSecondArray(
+            treatmentExceptionTypes,
+            treatmentTypes,
+        );
+
+        if (treatmentTypesAreValid) {
+            if (
+                coreFields.includes($name) &&
+                (!$field || $field === null || checkforEmpty($field))
+            ) {
+                return {
+                    valid: false,
+                    message: `The '${$name}' field must be submitted when the 'treatment_type' field is '${treatmentTypes}'`,
+                };
+            } else {
+                // console.log('test passed', $name, $field);
```

Comment:
unused else blcok should be removed

Reply:
left it there to indicate meaning/intent. will remove